packages/tracking-sdk/src/main.ts

```diff
@@ -598,13 +610,65 @@ export default function main() {
 
     if (!flags) {
       warn(`failed to fetch feature flags, using fall-back flags`);
-      return fallbackFlags.reduce((acc, flag) => {
+      flags = fallbackFlags.reduce((acc, flag) => {
         acc[flag.key] = flag;
         return acc;
       }, {} as Flags);
     }
 
-    return flags;
+    return proxify(flags, rateLimited(FLAG_EVENTS_PER_MIN, sendCheckEvent));
+  }
+
+  function sendCheckEvent(_: keyof Flag, flag: Flag) {
+    void featureFlagEvent({
+      action: "check",
+      flagKey: flag.key,
+      flagVersion: flag.version,
+      evalResult: flag.value,
+    });
+  }
+
+  /**
+   * Send a feature flags event.
+   *
+   *
+   * @param action The saction that was taken, either "evaluate" or "check".
+   *        "evaluate" is used when the flag is evaluated, and "check" is used when the flag is checked (after it has been evaluated).
+   * @param flagKey The key of the flag that was evaluated or checked.
+   * @param flagVersion The version of the flag that was evaluated or checked. Can be undefined if the flag is a fallback flag and version
+   *        is not known.
+   * @param evalResult The result of the flag evaluation.
+   * @param evalContext Optional context that was used to evaluate the flag.
+   * @param evalRuleResults Optional results of each rule that was evaluated.
+   * @param evalMissingFields Optional list of missing fields that were required to evaluate the flag.
+   */
+
+  async function featureFlagEvent(args: {
+    action: "evaluate" | "check";
+    flagKey: string;
+    flagVersion: number | undefined;
+    evalResult: boolean;
+    evalContext?: Record<string, any>;
+    evalRuleResults?: boolean[];
+    evalMissingFields?: string[];
+  }) {
+    checkKey();
+
+    const payload = {
+      action: args.action,
+      flagKey: args.flagKey,
+      flagVersion: args.flagVersion,
+      evalContext: args.evalContext,
+      evalResult: args.evalResult,
+      evalRuleResults: args.evalRuleResults,
+      evalMissingFields: args.evalMissingFields,
+    };
+
+    const res = await postRequest(makeUrl("flags/events"), payload);
```

Comment:
This sends the publishable key as a query param. The pattern should be:
- always use the “Authorization” header
- except for GET requests

Reply:
That would add a pre-flight for all events we send - `user`, `company` and `track` events since they are POSTs

Reply:
The ingest accepts POST body requests when they use the `Content-type: application/json` header. `Content-type: application/json` will trigger a preflight request regardless of the rest. So, we're already triggering pre-flight requests for those - and that's OK. There's no latency requirement for those, just for the feature flag requests.

<img width="788" alt="Screenshot 2024-07-10 at 10 26 25" src="https://github.com/bucketco/bucket-javascript-sdk/assets/744/c8f3a7a6-3444-443a-a050-fa09536fa12c">

Reply:
Updated the SDK to use the new style endpoints with the required params/headers